helm/prometheus-rules/templates/alerting-rules/kyverno.all.rules.yml

```diff
@@ -0,0 +1,76 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  creationTimestamp: null
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: kyverno.all.rules
+  namespace: {{ .Values.namespace  }}
+spec:
+  groups:
+  - name: webhooks
+    rules:
+    - alert: KyvernoWebhooksPresent
+      annotations:
+        description: 'Kyverno has no available replicas but webhooks are present.'
+        opsrecipe: kyverno-webhooks/
+      expr: sum(kube_validatingwebhookconfiguration_info{validatingwebhookconfiguration=~"kyverno-.*"}) > 0 and sum(kube_deployment_status_replicas{deployment=~"kyverno|kyverno-admission-controller"}) == 0
+      for: 5m
+      labels:
+        area: managedservices
+        cancel_if_cluster_status_creating: "true"
+        cancel_if_cluster_status_deleting: "true"
+        cancel_if_cluster_status_updating: "true"
+        cancel_if_outside_working_hours: "false"
+        severity: page
+        team: shield
+        topic: kyverno
+  - name: reports
+    rules:
+    - alert: KyvernoReportsTooHigh
+      annotations:
+        description: '{{`Kyverno {{ $labels.kind }} are too high.`}}'
+        opsrecipe: kyverno-reports/
+      expr: aggregation:kyverno_resource_counts{kind=~"(clusteradmissionreports|admissionreports).kyverno.io"} > 50000
+      for: 15m
+      labels:
+        area: managedservices
+        cancel_if_cluster_status_creating: "true"
+        cancel_if_cluster_status_deleting: "true"
+        cancel_if_cluster_status_updating: "true"
+        cancel_if_outside_working_hours: "false"
+        severity: page
+        team: shield
+        topic: kyverno
+  - name: replicas
+    rules:
+    - alert: KyvernoScaledDownTooLong
+      annotations:
+        description: 'Kyverno has been scaled down for too long.'
+        opsrecipe: kyverno-scaled-down/
+      expr: sum(kube_deployment_spec_replicas{deployment="kyverno"}) == 0
+      for: 1h
+      labels:
+        area: managedservices
+        cancel_if_cluster_status_creating: "true"
+        cancel_if_cluster_status_deleting: "true"
+        cancel_if_cluster_status_updating: "true"
+        cancel_if_outside_working_hours: "true"
+        severity: notify
+        team: shield
+        topic: kyverno
+    - alert: KyvernoWronglyScaled
+      annotations:
+        description: 'Kyverno deployment scaling has an unsupported amount of replicas'
+        opsrecipe: KyvernoWronglyScaled/
+      expr: sum(kube_deployment_spec_replicas{deployment="kyveno"}) > 0 && 3 <
```

Comment:
Probably something like 0 < sum < 3 was meant here? I'd hope that's valid promql syntax

Reply:
Not sure @Strigix , where you able to check this one?